relayer/relays/parachain/beefy-listener.go

```diff
@@ -258,8 +262,10 @@ func (li *BeefyListener) generateProof(ctx context.Context, input *ProofInput, h
 	// Generate a merkle proof for the parachain head with input ParaId
 	// and verify with merkle root hash of all parachain heads
 	// Polkadot uses the following code to generate merkle root from parachain headers:
-	// https://github.com/paritytech/polkadot/blob/2eb7672905d99971fc11ad7ff4d57e68967401d2/runtime/rococo/src/lib.rs#L706-L709
-	merkleProofData, err := CreateParachainMerkleProof(input.ParaHeads, input.ParaID)
+	// https://github.com/paritytech/polkadot-sdk/blob/d66dee3c3da836bcf41a12ca4e1191faee0b6a5b/polkadot/runtime/westend/src/lib.rs#L453-L460
+	// Truncate the ParaHeads to the 1024
+	// https://github.com/paritytech/polkadot-sdk/blob/d66dee3c3da836bcf41a12ca4e1191faee0b6a5b/polkadot/runtime/parachains/src/paras/mod.rs#L1305-L1311
+	merkleProofData, err := CreateParachainMerkleProof(input.ParaHeads[:MaxParaHeads], input.ParaID)
```

Comment:
So are we always generate proof against a static length array(i.e. 1024)? Or could we use `min(len(paraHeads),1024)`?

Reply:
Good catch! Added in [1d4c23a](https://github.com/Snowfork/snowbridge/pull/1288/commits/1d4c23a45c97563a828951ac35a06026e6b11f22).